lib/java/com/google/android/material/snackbar/BaseTransientBottomBar.java

```diff
@@ -1127,69 +1134,136 @@ protected SnackbarBaseLayout(@NonNull Context context, AttributeSet attrs) {
       animationMode = a.getInt(R.styleable.SnackbarLayout_animationMode, ANIMATION_MODE_SLIDE);
       backgroundOverlayColorAlpha =
           a.getFloat(R.styleable.SnackbarLayout_backgroundOverlayColorAlpha, 1);
-      setBackgroundTintList(
+      backgroundTint =
           MaterialResources.getColorStateList(
-              context, a, R.styleable.SnackbarLayout_backgroundTint));
-      setBackgroundTintMode(
+              context, a, R.styleable.SnackbarLayout_backgroundTint);
+      backgroundTintMode =
           ViewUtils.parseTintMode(
-              a.getInt(R.styleable.SnackbarLayout_backgroundTintMode, -1), PorterDuff.Mode.SRC_IN));
+              a.getInt(R.styleable.SnackbarLayout_backgroundTintMode, -1), PorterDuff.Mode.SRC_IN);
       actionTextColorAlpha = a.getFloat(R.styleable.SnackbarLayout_actionTextColorAlpha, 1);
+      if (a.hasValue(R.styleable.SnackbarLayout_android_background)) {
+        backgroundOverwritten = true;
+        ViewCompat.setElevation(this,snackbarElevation);
+        setBackgroundTintList(backgroundTint);
+      } else {
+        setInternalBackground(createDefaultShapeMaterialBackground());
+      }
       a.recycle();
 
       setOnTouchListener(consumeAllTouchListener);
       setFocusable(true);
 
-      if (getBackground() == null) {
-        ViewCompat.setBackground(this, createThemedBackground());
+    }
+
+    protected boolean isUsingOriginalBackground() {
+      return !backgroundOverwritten;
+    }
+
+    private Drawable createDefaultShapeMaterialBackground() {
+      materialBackgroundDrawable = new MaterialShapeDrawable(shapeAppearanceModel);
+      materialBackgroundDrawable.initializeElevationOverlay(getContext());
+      materialBackgroundDrawable.setElevation(snackbarElevation);
+      int backgroundDefaultColor =
+          MaterialColors.layer(
+              this, R.attr.colorSurface, R.attr.colorOnSurface, getBackgroundOverlayColorAlpha());
+      materialBackgroundDrawable.setFillColor(ColorStateList.valueOf(backgroundDefaultColor));
+      if (backgroundTint != null) {
+        DrawableCompat.setTintList(materialBackgroundDrawable, backgroundTint);
+      }
+      if (backgroundTintMode != null) {
+        DrawableCompat.setTintMode(materialBackgroundDrawable, backgroundTintMode);
       }
+      return materialBackgroundDrawable;
     }
 
     @Override
     public void setBackground(@Nullable Drawable drawable) {
       setBackgroundDrawable(drawable);
     }
 
+    void setInternalBackground(Drawable background) {
+      super.setBackgroundDrawable(background);
+    }
+
     @Override
     public void setBackgroundDrawable(@Nullable Drawable drawable) {
-      if (drawable != null && backgroundTint != null) {
-        drawable = DrawableCompat.wrap(drawable.mutate());
-        DrawableCompat.setTintList(drawable, backgroundTint);
-        DrawableCompat.setTintMode(drawable, backgroundTintMode);
+      if (isUsingOriginalBackground()){
```

Comment:
Don't really think we should have the if-else clauses here. 

We will always need L1207~1212 to be executed I think.